app/src/Containers/UserForm/index.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useRef, useState } from "react";
+import { passwordValidation } from "../../Helpers/password-validation";
+import { UserFormWrapper } from "./styles";
+import qs from "qs"
+
+interface user {
+  username: string,
+  password: string,
+}
+
+const reqUserCreate = async (newUser: user) => {
+  const in_flight = await fetch("/api/user/create", {
+    method: 'POST',
+    body: JSON.stringify(newUser),
+    headers: {
+      'Content-Type': "application/json"
+    }
+  });
+  const output = await in_flight.json();
+  return output
+}
+
+const reqLogin = async (user: user) => {
+  const in_flight = await fetch('/api/user/login', {
+    method: 'POST',
+    body: JSON.stringify(user),
+    headers: {
+      'Content-type': 'application/json'
+    }
+  })
+  const output = await in_flight.json()
+  return output
+}
+
+const creationErrorHandler = (res: any) => {
+  if(res.code === 11000){
+    alert("That username is already in use.")
+    return false
+  }
+}
+
+const loginErrorHandler = (res: any) => {
+  alert(res)
+}
+
+const UserForm = (props: any) => {
+  
+  const [newAccount, setNewAccount] = useState(false)
+  const usernameInput = useRef()
+  const passwordInput = useRef()
+  const passwordCheckInput = useRef()
```

Comment:
no ref, use setState and callbacks. a ref should only be used if you gotta manipulate the DOM for some reason. you don't need to modify the DOM here, it's only displaying data. for this situation, because the actual DOM isn't providing any info, for instance:

`imagine needing to know where an element is rendered in relation to another element`

this can only be found out after a render has occurred, as internally react keeps track of the element 'values' but none of the post render layout, which `refs` give you access to